tpu-client/src/tpu_connection_cache.rs

```diff
@@ -245,27 +255,22 @@ impl<P: ConnectionPool> TpuConnectionCache<P> {
     }
 }
 
-impl<P: ConnectionPool> Default for TpuConnectionCache<P> {
-    fn default() -> Self {
-        Self {
-            map: RwLock::new(IndexMap::with_capacity(MAX_CONNECTIONS)),
-            stats: Arc::new(ConnectionCacheStats::default()),
-            last_stats: AtomicInterval::default(),
-            connection_pool_size: DEFAULT_TPU_CONNECTION_POOL_SIZE,
-            tpu_config: P::TpuConfig::default(),
-        }
-    }
-}
-
 #[derive(Error, Debug)]
 pub enum ConnectionPoolError {
     #[error("connection index is out of range of the pool")]
     IndexOutOfRange,
 }
 
+pub trait FallibleCtor {
+    type ClientError;
+    fn new() -> Result<Self, Self::ClientError>
```

Comment:
Open to bikeshedding on this trait name, and whether the method should be a variation of new vs a variation of default.

Reply:
it should clearly be called `Defallt`

Reply:
Genius!

Reply:
joking aside. wdyt of just calling it something like `TpuConfigNew` with the same named method, just snake-case?

Reply:
Wfm! I'm going to make it NewTpuConfig, though